Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -227,14 +235,20 @@ Handler Launcher::get_handler_for_executable(Handler::Type handler_type, String
     return handler;
 }
 
-bool Launcher::open_with_user_preferences(HashMap<String, String> const& user_preferences, String const& key, Vector<String> const& arguments, String const& default_program)
+bool Launcher::open_with_user_preferences(HashMap<String, String>& user_preferences, String const& key, Vector<String> const& arguments, String const& default_program)
 {
-    auto program_path = user_preferences.get(key);
-    if (program_path.has_value())
-        return spawn(program_path.value(), arguments);
+    String executable = "";
+    if(for_each_handler(key, user_preferences, [&](auto const& handler) -> bool {
```

Comment:
The `if(` looks incorrectly formatted, did you run `clang-format` on the code?

Btw, we have commit hooks that automatically run certain checks, including the formatting of the code, see: https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#commit-hooks

Reply:
Oops, I just installed the pre-commit hooks, and fixed formatting in the latest commit.